app/src/test/kotlin/org/cru/godtools/ui/dashboard/lessons/LessonsViewModelTest.kt

```diff
@@ -62,9 +95,50 @@ class LessonsViewModelTest {
         val second = randomTool("second", Tool.Type.LESSON, defaultOrder = 2, isHidden = false)
 
         viewModel.lessons.test {
-            lessonsFlow.value = listOf(second, first)
+            lessonsByLanguageFlow.value = listOf(second, first)
             runCurrent()
             assertEquals(listOf("first", "second"), expectMostRecentItem())
         }
     }
+
+    @Test
+    fun `Property lessons - Filter by language`() = testScope.runTest {
+        val first = randomTool("first", Tool.Type.LESSON, defaultOrder = 1, isHidden = false)
+        val second = randomTool("second", Tool.Type.LESSON, defaultOrder = 2, isHidden = false)
+
+        viewModel.lessons.test {
+            viewModel.updateSelectedLanguage(Language(Locale.ENGLISH))
```

Comment:
you should verify that the repository is being properly queried with whatever your current selected language is. Take a look at `coVerify {}` for how to do it